File: mytime-api/app/models/employee_document.py
from sqlalchemy import Column, BigInteger, String, Boolean, DateTime, Text
from sqlalchemy.dialects.mssql import DATETIMEOFFSET
from app.core.database import Base


class EmployeeDocument(Base):
    __tablename__ = "EmployeeDocuments"

    EmployeeDocumentId = Column(
        BigInteger,
        primary_key=True,
        index=True,
        autoincrement=True
    )
    EmployeeId = Column(BigInteger, nullable=True)
    DocumentType = Column(String(255), nullable=True)
    FileId = Column(String(255), nullable=True)
    FileName = Column(String(500), nullable=True)
    BucketId = Column(String(255), nullable=True)
    ContentLength = Column(BigInteger, nullable=True)
    ContentType = Column(String(255), nullable=True)
    FileInfo = Column(Text, nullable=True)
    UploadTimestamp = Column(DateTime, nullable=True)
    CreatedOn = Column(DateTime, nullable=True)
    CreatedBy = Column(BigInteger, nullable=True)
    ModifiedOn = Column(DateTime, nullable=True)
    ModifiedBy = Column(BigInteger, nullable=True)
    IsActive = Column(Boolean, nullable=True)

    def __repr__(self):
        return f"<EmployeeDocument(EmployeeDocumentId={self.EmployeeDocumentId}, FileName='{self.FileName}')>"

File: mytime-ui/src/app/admin/models/employee_document.ts
export interface EmployeeDocument {
  EmployeeDocumentId: number;
  EmployeeId?: number | null;
  DocumentType?: string | null;
  FileId?: string | null;
  FileName?: string | null;
  BucketId?: string | null;
  ContentLength?: number | null;
  ContentType?: string | null;
  FileInfo?: string | null;
  UploadTimestamp?: string | null;
  CreatedOn?: string | null;
  CreatedBy?: number | null;
  ModifiedOn?: string | null;
  ModifiedBy?: number | null;
  IsActive?: boolean | null;
}
